addons/mail/static/src/messaging/entity/device/device.js

```diff
@@ -0,0 +1,94 @@
+odoo.define('mail.messaging.entity.Device', function (require) {
+'use strict';
+
+const { registerNewEntity } = require('mail.messaging.entity.core');
+
+const config = require('web.config');
+
+function DeviceFactory({ Entity }) {
+
+    class Device extends Entity {
+
+        start() {
+            this.constructor._listenGlobalWindowResize(this);
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * Method helps overriding in tests.
+         *
+         * @private
+         * @static
+         */
+        static _listenGlobalWindowResize(entity) {
+            window.addEventListener('resize', _.debounce(() => entity.update()), 100);
```

Comment:
env.window, move this code in start, and adapt env instead of patching entity in test